Emby.Naming/Video/VideoListResolver.cs

```diff
@@ -209,5 +264,41 @@ private static bool IsEligibleForMultiVersion(ReadOnlySpan<char> folderName, Rea
                    || testFilename[0] == '-'
                    || CheckMultiVersionRegex().IsMatch(testFilename);
         }
+
+        private static string EpisodeGrouper(string testFilePath, NamingOptions namingOptions, CollectionType? collectionType)
+        {
+            // Grouper for tv shows/episodes should be everything before space-dash-space
+            var resolver = new EpisodeResolver(namingOptions);
+            var episodeInfo = resolver.Resolve(testFilePath, false);
+            ReadOnlySpan<char> seriesName = episodeInfo!.SeriesName;
+
+            var filename = Path.GetFileNameWithoutExtension(testFilePath);
+            // Start with grouping by filename
+            string g = filename;
+            for (var i = 0; i < namingOptions.VideoVersionRegexes.Length; i++)
+            {
+                var rule = namingOptions.VideoVersionRegexes[i];
+                var match = rule.Match(filename);
+                if (!match.Success)
+                {
+                    continue;
```

Comment:
(not a Jellyfin developer)

It is possible that more than one regex will exist in the future, but the current code is still looping over next regexes even if it matched. Why not rewriting it like this:

```cs
if (match.Success)
{
    g = match.Groups["filename"].Value;
    // ...
    // insert the rest of the code below
    // ...
    break;
}
```

An alternative would be use this form:

```cs
for (var i = 0; i < namingOptions.VideoVersionRegexes.Length; i++)
{
    var rule = namingOptions.VideoVersionRegexes[i];
    var match = rule.Match(filename);
    if (match.Success)
    {
        string g = match.Groups["filename"].Value;
        // ...
        // rest of the code
        // ...
        return g;
    }
}
return filename;
```